enterprise/tailnet/pgcoord.go

```diff
@@ -632,6 +622,20 @@ func (b *binder) retrieveBinding(bk bKey) binding {
 	return bnd
 }
 
+func (b *binder) wait() {
+	b.workerWG.Wait()
+
+	b.logger.Debug(b.ctx, "binder exiting, updating peers to lost", slog.Error(b.ctx.Err()))
+
+	err := b.store.UpdateTailnetPeerStatusByCoordinator(context.Background(), database.UpdateTailnetPeerStatusByCoordinatorParams{
```

Comment:
A little too clever to put this here. It's surprising in a way that could lead to bugs, because calling `wait()` implies that you're just doing nothing waiting for something else to finish, but here it has a side effect.  `wait()` could get called 0 times, or more than once.  We should call this, exactly once, whenever we shut down the binder, independently of whether someone chooses to wait for the shut down to complete.

To do that, put this call in a goroutine we start after the workers, that waits for them to finish.  You'll need to track it independently of the `workerWG`, lest we deadlock.